statement_test.go

```diff
@@ -0,0 +1,184 @@
+package gosnowflake
+
+import (
+	"database/sql"
+	"testing"
+)
+
+func TestMultiStatementTransaction(t *testing.T) {
+	var db *sql.DB
+	var err error
+	ndsn, err := AddConnParameter(dsn, MultiStatementCount, "0")
+	if err != nil {
+		t.Fatalf("failed to add connection parameter. err: %v", err)
+	}
+	if db, err = sql.Open("snowflake", ndsn); err != nil {
+		t.Fatalf("failed to open db. %v, err: %v", dsn, err)
+	}
+	defer db.Close()
+	_, err = db.Exec("drop table if exists test_multi_statement_txn")
+	if err != nil {
+		t.Fatalf("failed to drop table: %v", err)
+	}
+
+	_, err = db.Exec("create or replace table test_multi_statement_txn(c1 number, c2 string)" +
+		"as select 10, 'z'")
+	if err != nil {
+		t.Fatalf("failed to create table: %v", err)
+	}
+
+	multiStmtQuery := "begin;\n" +
+		"delete from test_multi_statement_txn;\n" +
+		"insert into test_multi_statement_txn values (1, 'a'), (2, 'b');\n" +
+		"commit;"
+	db.Exec(multiStmtQuery)
+
+	_, err = db.Exec("drop table if exists test_multi_statement_txn")
+	if err != nil {
+		t.Fatalf("failed to drop table: %v", err)
+	}
+}
+
+func TestMultiStatementQuery(t *testing.T) {
+	var db *sql.DB
+	var err error
+	ndsn, err := AddConnParameter(dsn, MultiStatementCount, "0")
+	if err != nil {
+		t.Fatalf("failed to add connection parameter. err: %v", err)
+	}
+	if db, err = sql.Open("snowflake", ndsn); err != nil {
+		t.Fatalf("failed to open db. %v, err: %v", dsn, err)
+	}
+	defer db.Close()
+
+	multiStmtQuery := "select 123;\n" +
+		"select 456;\n" +
+		"select 789;\n" +
+		"select '000';"
+
+	rows, _ := db.Query(multiStmtQuery)
```

Comment:
Can you please revisit the Query interface why it cannot take extra custom parameter? I thought you could add it. If not, did you consider using `Context`?

Reply:
The goal is to set the statement level parameter for multistatement support. Did you explore any other option?

Reply:
Upon research, I will move forward with context. The Query interface cannot take other parameters, making it an invalid interface. We can create a wrapper function such as QueryContextMulti() of some sort, however.